src/api/recipes.ts

```diff
@@ -45,3 +46,13 @@ export function useFetchRecipeById(id?: string) {
     },
   });
 }
+
+export function usePostRecipe() {
+  return useMutation<void, FirebaseError, Recipe>({
+    mutationFn: async (newRecipe: Recipe) => {
+      await addDoc(recipeCollection, {
+        ...newRecipe,
+      });
```

Comment:
If I've seen this in code I wouldn't question it as it seems ok and I didn't work much with react query, but I wonder if you need to spread newRecipe to new object if there are no other props passed. Is it mutated in some way?
Is there a reason to create new object and pass it here?

This won't change much as it will be very small memory optimalization so it won't really matter, but I suggest you look into it as a practise if you have time for it